### PR TITLE
[Docker][Documentation] Uses correct docker image from Docker Hub

### PR DESCRIPTION
**Pull request by ghost**
This PR uses the correct docker image from [Docker Hub](https://registry.hub.docker.com/u/sytse/gitlab-data/) for the `gitlab-data` image.

The PR also avoids using ___ in image tag names wherever it is possible.

More information under GitLab issue #9364.

/cc @dosire.

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -72,15 +72,15 @@ sudo docker run -ti -e TERM=linux --name gitlab-ce-troubleshoot --publish 8080:8
 ### Get published images from Dockerhub
 
 ```bash
-sudo docker pull sytse/gitlab-data
+sudo docker pull sytse/gitlab_data
```
Comment (dosire) [inline docker/README.md]: Shouldn't this be with a hyphen as well?

Comment (stefan-it) [inline docker/README.md]: Then you have to rename the image at docker-hub. A docker pull won't work with `sytse/gitlab-data` at the moment.

Comment (dosire) [inline docker/README.md]: Yes, after this MR is merged I'll upload the new image.

Comment (stefan-it) [inline docker/README.md]: Great :) I changed the image name back to `sytse/gitlab-data` and squashed the commits! I hope everything is fine now :)

**Comment by dosire** [comment]
I've uploaded gitlab-data and deleted gitlab_data from dockerhub.

**Pull request merged**